api/src/main/java/module-info.java

```diff
@@ -638,39 +631,37 @@
  * <td style="vertical-align: top"><b>Notes</b></td>
  * </tr>
  *
- * <tr style="vertical-align: top; background-color:#eee"><td>{@code countBy...}</td>
+ * <tr style="vertical-align: top; background-color:#eee"><td>{@code count}</td>
  * <td>{@code long},
  * <br>{@code int}</td>
  * <td></td></tr>
  *
- * <tr style="vertical-align: top"><td>{@code deleteBy...},
- * <br>{@code updateBy...}</td>
+ * <tr style="vertical-align: top"><td>{@code delete}</td>
  * <td>{@code void},
- * <br>{@code boolean},
```

Comment:
A boolean return value for delete seems convenient for a query that can only possibly delete one entity, such as deletion by unique identifier, so it would be nice to keep it unless it is causing trouble or inconsistency with JDQL delete, in which case I suppose Jakarta Data providers could allow it on their own rather than as a required type.

Reply:
So, what do we say for other kinds of delete?

In ch5, for JDQL:

> A `delete` statement ... deletes each record which satisfies the restriction imposed by the `where` clause, and returns the number of deleted records to the client.

In the javadoc for `@Delete`:

> The method return type must be `void`, `int`, or `long`.

In the javadoc for `@Query`:

> For `update` or `delete` statements, it must be `void`, `int` or `long`.

So I think we should stay consistent here.

(As usual, it doesn't mean implementations can't experiment with allowing other things. We can always add more flexibility later.)